frameworks/moveos-stdlib/sources/linked_table.move

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+/// Similar to `sui::table` but the values are linked together, allowing for ordered insertion and
+/// removal
+module moveos_std::linked_table {
+    use std::option::{Self, Option};
+    use moveos_std::object::Object;
+    use moveos_std::object;
+
+
+
+    // Attempted to destroy a non-empty table
+    const ErrorTableNotEmpty: u64 = 0;
+    // Attempted to remove the front or back of an empty table
+    const ErrorTableIsEmpty: u64 = 1;
+
+    struct LinkedTable<K: copy + drop + store, phantom V: store> has key, store {
+        /// the number of key-value pairs in the table
+        size: u64,
+        /// the front of the table, i.e. the key of the first entry
+        head: Option<K>,
+        /// the back of the table, i.e. the key of the last entry
+        tail: Option<K>,
```

Comment:
We should put a `handle: Object<TablePlaceholder>` field in the LinkedTable and store the value using the handle Object.